### PR TITLE
 bam2fasta package to simplify sourmash compute

### PR DESCRIPTION
**Pull request by pranathivemuri**
- Removes bam2fasta conversion bits from sourmash compute
- adds bam2fasta dependency
- names tests separated out when input-is-10x
- adds bam2fasta pip dependency and removes pathos, pysam

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by pranathivemuri** [comment]
@olgabot @ctb @luizirber Please review when you are free! Here is the pip package I released for bam2fasta conversion - https://pypi.org/project/bam2fasta/

File: sourmash/command_compute.py [hunk 1]
```diff
@@ -72,11 +66,11 @@ def compute(args):
                         help="name the signature generated from each file after the first record in the file (default: False)")
     parser.add_argument('--input-is-10x', action='store_true',
                         help="Input is 10x single cell output folder (default: False)")
-    parser.add_argument('--count-valid-reads', default=0, type=int,
+    parser.add_argument('--min-umi-per-barcode', default=0, type=int,
```
Comment (luizirber) [inline sourmash/command_compute.py]: Changing arguments require major version bumps. In this case this argument is already available in sourmash 2.2.0, so we need to revert it to the old name or adapt the call to bam2fasta (convert a `--count-valid-reads` argument in sourmash to `--min-umi-per-barcode` argument in `bam2fasta` at https://github.com/dib-lab/sourmash/pull/768/files#diff-56040d40281982d44bcd65f91eb0b132R280)

File: setup.py [hunk 2]
```diff
@@ -72,7 +72,7 @@
         'test' : ['pytest', 'pytest-cov', 'numpy', 'matplotlib', 'scipy', 'recommonmark'],
         'demo' : ['jupyter', 'jupyter_client', 'ipython'],
         'doc' : ['sphinx'],
-        '10x': ['pathos', 'pysam']
+        '10x': ['bam2fasta']
```
Comment (luizirber) [inline setup.py]: We can specify the dependency here as `bam2fasta==1.0.1`, and update as needed in the future.

**Comment by luizirber** [comment]
We also need to get `bam2fasta` into bioconda before releasing a new version of sourmash.

**Comment by pranathivemuri** [comment]
@luizirber do you have a config file/recipe file you used to release sourmash on bioconda. I have released the pip bam2fasta packages, thanks to release.md instructions in sourmash/docs. I saw the bioconda's autobump feature as one of the steps, but it didn't have much details how to create a recipe there.

**Comment by pranathivemuri** [comment]
@luizirber addressed your comments, thanks!

**Comment by luizirber** [comment]
> @luizirber addressed your comments, thanks!

Wow, that was fast! =]

As a rule of thumb, if you need to change a test it's probably a breaking change for the API. That's one reason why we never remove tests, and avoid changing them as much as possible...

> do you have a config file/recipe file you used to release sourmash on bioconda. I have released the pip bam2fasta packages, thanks to release.md instructions in sourmash/docs. I saw the bioconda's autobump feature as one of the steps, but it didn't have much details how to create a recipe there.

I usually start from a previous recipe that was recently updated. The official documentation is https://bioconda.github.io/contributor/index.html, and the linked https://github.com/bioconda/bioconda-recipes/blob/master/recipes/pyfaidx/meta.yaml is a pretty good start. It's also a good idea to add yourself as a maintainer (see https://github.com/bioconda/bioconda-recipes/blob/faf2f3ecbc2093cec7237974683f8bcdb64d1e47/recipes/salmon/meta.yaml#L41 for an example)

**Comment by pranathivemuri** [comment]
@luizirber bam2fasta can be downloaded via bioconda now, just tried it! Thanks for the comments/leads on how to start the bioconda package.  Thanks to your directions and the release.md in sourmash repo, I can put pip and bioconda packages in my skills bucket now! :D

Please review this PR when you are free!

**Pull request merged**